runtime/lua/vim/filetype.lua

```diff
@@ -1007,6 +1036,156 @@ local extension = {
       return 'text'
     end
   end,
+  cmd = function(path, bufnr)
+    if getline(bufnr, 1):find('^/%*') then
+      return 'rexx'
+    end
+    return 'dosbatch'
+  end,
+  rul = function(path, bufnr)
+    require('vim.filetype.detect').rul(bufnr)
+  end,
+  cpy = function(path, bufnr)
+    if getline(bufnr, 1):find('^##') then
+      return 'python'
+    end
+    return 'cobol'
+  end,
+  dsl = function(path, bufnr)
+    if getline(bufnr, 1):find('^%s*<!') then
+      return 'dsl'
+    end
+    return 'structurizr'
+  end,
+  edf = 'edif',
+  edfi = 'edif',
+  edo = 'edif',
+  edn = function(path, bufnr)
+    if getline(bufnr, 1):find('^%s*%(%s*edif>') then
+      return 'edif'
+    end
+    return 'clojure'
+  end,
+  smil = function(path, bufnr)
+    if getline(bufnr, 1):find('<%?%s*xml.*%?>') then
+      return 'xml'
+    end
+    return 'smil'
+  end,
+  smi = function(path, bufnr)
+    if getline(bufnr, 1):find('<smil>') then
+      return 'smil'
+    end
+    return 'mib'
+  end,
+  install = function(path, bufnr)
+    if getline(bufnr, 1):find('<%?php') then
+      return 'php'
+    end
+    vim.fn['dist#ft#SetFileTypeSH']('bash')
+  end,
+  pm = function(path, bufnr)
+    local line = getline(bufnr, 1)
+    if line:find('XPM2') then
+      return 'xpm2'
+    elseif line:find('XPM') then
+      return 'xpm'
+    else
+      return 'perl'
+    end
+  end,
+  me = function(path, bufnr)
+    local filename = vim.fn.fnamemodify(path, ':t')
```

Comment:
This should have a `:lower()` at the end because the behavior of `!=` depends on the `ignorecase` option just like `~=`